src/pybind/mgr/orchestrator.py

```diff
@@ -592,6 +592,9 @@ def __init__(self, host_pattern, data_devices, db_devices=None, wal_devices=None
         #: To fully utilize nvme devices multiple osds are required.
         self.osds_per_device = osds_per_device
 
+        #: A list of strings, containing paths which should back OSDs
+        self.data_directories = data_directories
```

Comment:
could you add a test to `validate()` that `data_directories` and `data_devices` are mutually exclusive?

Reply:
I don't think they need to be mutually exclusive. You can pass it a DriveGroupSelection with both data_directories and data_devices in it and rook can create OSDs for all of them.